### ISSUE TITLE
Not showing in real device

### ISSUE THREAD

**Issue by xxx44yyy** [ratson/cordova-plugin-admob-free#307]
After run example app (https://github.com/ratson/cordova-plugin-admob-free/tree/master/examples) on real device, ads not showing.  isTesting: true and autoShow: true included. 

Is that right?

**Comment by ConcaveIT** [comment]
Same case for me, real ads not showing in device, please somebody tell me why?

**Comment by Winamp9** [comment]
did according to the instructions?  https://developers.google.com/admob/android/quick-start

**Comment by ConcaveIT** [comment]
Yes, Real Ads showing on ionic DevApp but not in production apk.

**Comment by xxx44yyy** [comment]
Emulator show production and test ads. Real device does not show anything.

@Winamp9 What can help guide for Android? We are trying to use Cordova :)

**Comment by ConcaveIT** [comment]
Is this worked for you? And have you publish your app to play store? @xxx44yyy

**Comment by xxx44yyy** [comment]
I can't find a solution to the problem, so I'll probably try to publish the app. But not today, maybe tomorrow.

**Comment by ConcaveIT** [comment]
I haven't play store account yet. may be that should work. Please let me know if you find any solution, (same for me also)

**Comment by xxx44yyy** [comment]
Yeah, okay. I will still look for a solution and if I do not find it, I will have to try to publish it.

**Comment by xxx44yyy** [comment]
@ConcaveIT It worked. I added this to the configuration:
`"cordova-admob-sdk": "0.20.0"`

and this

```
    <plugin name="cordova-admob-sdk" spec="../plugins/cordova-admob-sdk">
        <variable name="PLAY_SERVICES_VERSION" value="+" />
    </plugin>

```
After that, a banner (test ads) appeared on the real phone.

Then, I published the application and it appeared real advertising (of course, the code has already turned off the parameter isTesting). 
https://play.google.com/store/apps/details?id=com.a2_finance.cordova.test.ads
I think Google will swear that I published such an application, so I will delete it today.

You can download the APK (https://apkpure.com/ru/cordova-ads-test/com.a2_finance.cordova.test.ads), and decompile it.

**Comment by xxx44yyy** [comment]
To see real advertising - apparently Yes.

**Comment by yacinknn** [comment]
>  `"cordova-admob-sdk": "0.20.0"`

where i can add this ?
 "cordova-admob-sdk": "0.20.0"

**Comment by yacinknn** [comment]
bro thank u it worked for me
but i have 2 questions :
1-the banner ads display But not always ( Only sometimes ) why ?
2- i want to display  interstitial ads but automaticaly ( like in plugin of admobpro )  how can i do it?
thank u very much

**Comment by ConcaveIT** [comment]
@Yacinknn, can you share your code?

**Comment by ConcaveIT** [comment]
Confusion about one thing  some apps does not belongs to play store but they shows admob ad. How?

**Comment by xxx44yyy** [comment]
@ConcaveIT They're probably using test IDs. https://developers.google.com/admob/android/test-ads

**Comment by ConcaveIT** [comment]
@xxx44yyy,
No, because they do clicking business.
Other thing, you can check removing apps from play store, and see what happens.

**Comment by xxx44yyy** [comment]
@ConcaveIT To create an ad unit in Admob, you need an ID from GooglePlay. Therefore, I think if you remove the application from GooglePlay, then nothing will change, because in Admob the application will be registered.

But it doesn't matter now. The important thing is that:
a) test advertising is shown (on emulator - **Yes**, on real device - **Yes**)
b) real advertising is shown (on emulator - **No**, on real device - **Yes**, _after publication in Google Play_)

**Comment by diegoestacho999** [comment]
What info do I get from you? Here I am using and running all ad methods.

**Comment by diegoestacho999** [comment]
By the tests I've done so far, in the emulator only test ads same. In the actual device everything okay, they exhibit all without problems.
Try to use something like:

ionic cordova run android --prod

**Comment by darkthsideous** [comment]
Hello @yacinknn @xxx44yyy you guys are life savers thank you. I was wondering where in config.xml you add `"cordova-admob-sdk": "0.20.0"` in the <plugin />  tags? Thank you much

**Comment by xxx44yyy** [comment]
@darkthsideous not config.xml. In **package.json**. It's mine:

```
{
    "name": "cordovaads",
    "displayName": "CordovaAds",
    "version": "1.0.0",
    "description": "A sample cordova app for testing ads.",
    "main": "index.js",
    "scripts": {
        "test": "echo \"Error: no test specified\" && exit 1"
    },
    "dependencies": {
        "cordova-admob-sdk": "0.20.0",
        "cordova-android": "^7.1.1",
        "cordova-browser": "^5.0.4",
        "cordova-ios": "^4.5.5",
        "cordova-plugin-admob-free": "^0.21.0",
        "cordova-plugin-whitelist": "^1.3.3",
        "cordova-promise-polyfill": "0.0.2"
    },
    "cordova": {
        "plugins": {
            "cordova-plugin-whitelist": {},
            "cordova-plugin-admob-free": {
                "ADMOB_APP_ID": "ca-app-pub-XXXXXXXXXXX~YYYYYYYYYYY"
            }
        },
        "platforms": [
            "ios",
            "android",
            "browser"
        ]
    }
}
```

**Comment by darkthsideous** [comment]
@xxx44yyy  oh! that makes perfect sense. Thank you so much

**Comment by yacinknn** [comment]
thank u darkthsideous 
i ask a question about admob if i use interstitial auto Show =true
is it legal for admob . i afraid that i will be banned from admob if i use it !!

**Comment by yacinknn** [comment]
brp

> 
> 
> @yacinknn sorry for late reply. I dont believe you will get banned i think it only records how many times you use interstitial per impression to avoid spamming. I use autoshow= true, after a certain page in my app loads. i havent had any problems yet

bro the problem of autoshow = true is showing interstitial  ads after every seconds , and this is not allowed in terms of admob ( i have exeperience with this issue ) , can u tell me how to use autoshow= true, after a certain page in ur app loads.? thank u very much bro

**Comment by darkthsideous** [comment]
Yeah that is definitely not allowed for it to continuously show.

i usually just have my generic launch interstitial function and call it in ionVIewDidLoad

`launchInterstitial() {

    let interstitialConfig: AdMobFreeInterstitialConfig = {
      autoShow: true,
      id: 'ca-app-pub-AD-Mob-ID'
    };

    this.admob.interstitial.config(interstitialConfig);

    this.admob.interstitial.prepare().then(() => {
      // success
    });

  }`

**Comment by darkthsideous** [comment]
@xxx44yyy i tried your fix same issue occurring for me. Basically my ads are not showing at not a blank banner just completely gone as if it didn't launch at all. however it is launching in my console.log thoughts?

**Comment by xxx44yyy** [comment]
@darkthsideous I think you should create a new issue.

**Comment by darkthsideous** [comment]
@xxx44yyy ugh that bad. ok i will thank you. Just to confirm what version of the plugin are you using. Also have you tried admob-plus

**Comment by satyam-dev** [comment]
Anyone found a solution? Man o man I've tried everything from cleaning the cache to removing platforms and reinstalling them, deleting the node modules, deleting plugins folder, deletion node modules and re adding all of them. Nothing works!! On real device it does not work and while I test it on DevsApp, it works damn smooth with all the banner and Interstitial ads! Please help

**Comment by EricChuddy** [comment]
@xxx44yyy
please I have done everything but it's still nothing display anything, could it be that I am not getting the javascript code? If yes, please how will the javascript be like and where will I place it? I am using visual studio 2017

**Comment by darkthsideous** [comment]
@EricChuddy have you seen anymore active issues. Im afraid you might be facing a different issue

**Comment by joy2012bd** [comment]
> To see real advertising - apparently Yes.

When I set isTesing:true; ads is showing in real device.But if I set it false or comment out like // isTesing: true; ads is not showing. Do I Need to Publish It in Play Store? Otherwise ads won't be showing?

**Comment by EricChuddy** [comment]
@darkthsideous
It shows me error when I try to build

**Comment by mayekarakshay24** [comment]
@xxx44yyy ads are not shown when i tried to create app with cordova.i even created app with phonegap still it is not working. can u plz help i dont understand whats the problem.if this issue i solved it will be great help from u

**Comment by saggittarium** [comment]
If you want to debug your setup for android, this may help:

You will find Java lib source:
platforms\android\app\src\main\java\name\ratson\cordova\admob\Admob.java
try adding some logs/flags and track them in the logcat logs.

Also:
https://stackoverflow.com/questions/45069642/why-remove-admobs-addtestdevice-in-production

Cheers
